src/bindgen/library.rs

```diff
@@ -172,6 +175,39 @@ impl Library {
         &self.config
     }
 
+    fn erase_transparent_types_for_items<T: Item + Clone>(
+        &self,
+        eraser: &mut TransparentTypeEraser,
+        items: &ItemMap<T>,
+    ) -> ItemMap<T> {
+        // NOTE: Because `items` is actually a shared reference to `self`, we cannot take it as
+        // mutable. We also cannot `drain` or `take` it first, because then the items would be
+        // unavailable for lookup during the type erasure process. So we mutate a clone, and let the
+        // caller assign the result back after these shared references have died.
```

Comment:
This also looks really expensive for big projects... I'd much rather find a way to avoid this unconditional clone of everything... That may mean rejiggering how we store the items, or at the very least changing how we do the pass.

It seems the only reason we need the library around is to do the `transparent_alias` stuff, correct?

If so, would it make sense to first do a pass over `self.opaque_items`, `self.typedefs`, and `self.structs`, in order to collect a `Path -> Type` map, and only pass _that_ to `erase_transparent_types`?